File: BE/Project.Net8/Models/Major/BaiThiModel.cs
using BaiThi.Models.Major;
using DTC.DefaultRepository.Constants;
using DTC.DefaultRepository.Models.Base;
using DTC.T;
using MongoDB.Bson.Serialization.Attributes;
using Project.Net8.Models.Core;
using CoreModel = DTC.DefaultRepository.Models.Core.CoreModel;

namespace Project.Net8.Models.Major;

public class BaiThiModel: Audit, TEntity<string>
{

    public int DienTich { get; set; }
    public int GiaTriNen { get; set; } 
    public CommonModelShort TrangThai { get; set; }

    public string DiaChi { get; set; }
    public string GhiChu { get; set; }
    public string TenNguoiMua { get; set; }
    public string CMND { get; set; }
    public DateTime? NgayKyHopDong { get; set; }

    [BsonIgnore]
    public string NgayKyHopDongShow
    {
        get { return NgayKyHopDong.HasValue ? NgayKyHopDong.Value.ToLocalTime().ToString(FormatTime.FORMAT_DATE_CORE) : ""; }
    }

    public string SDT { get; set; }

}

File: FE/src/models/baiThiModel.js
import moment from "moment";
const toJson = (item) => {
  return {
    id: item.id,
    name: item.name,
    dienTich: item.dienTich,
    giaTriNen: item.giaTriNen,
    trangThai: item.trangThai,
    diaChi: item.diaChi,
    ghiChu: item.ghiChu,

    tenNguoiMua: item.tenNguoiMua,
    sdt: item.sdt,
    cmnd: item.cmnd,
    ngayKyHopDong: item.ngayKyHopDong,
    ngayKyHopDongShow: item.ngayKyHopDongShow,
  };
};

const fromJson = (item) => {
  return {
    id: item.id,
    name: item.name,
    dienTich: item.dienTich,
    giaTriNen: item.giaTriNen,
    trangThai: item.trangThai,
    diaChi: item.diaChi,
    ghiChu: item.ghiChu,

    tenNguoiMua: item.tenNguoiMua,
    sdt: item.sdt,
    cmnd: item.cmnd,
    ngayKyHopDong: item.ngayKyHopDong,
    ngayKyHopDongShow: item.ngayKyHopDongShow,
  };
};

const baseJson = () => {
  return {
    id: null,
    name: null,
    dienTich: 0,
    giaTriNen: 0,
    trangThai: null,
    diaChi: null,
    ghiChu: null,

    tenNguoiMua: null,
    sdt: null,
    cmnd: null,
    ngayKyHopDong: null,
    ngayKyHopDongShow: null,
  };
};

const toListModel = (items) => {
  if (items.length > 0) {
    let data = [];
    items.map((value, index) => {
      data.push(fromJson(value));
    });
    return data ?? [];
  }
  return [];
};
export const baiThiModel = {
  toJson,
  fromJson,
  baseJson,
  toListModel,
};
